### PR TITLE
fix: #1719

### PR DESCRIPTION
**Pull request by orange4glace**
- [x] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

fix for #1719 

When sending a 'ok' and 'invalid' message, compiler name is passed as a data.
compiler name can be defined in config options. If user does not provided it, the script generates a random guid as a compiler name.

### CODE REVIEW

File: client-src/default/index.js [hunk 1]
```diff
@@ -9,6 +9,32 @@ const { log, setLogLevel } = require('./utils/log');
...
+    urlParts = url.parse(scriptHost || '/', false, true);
+  }
+  const { auth, path } = urlParts;
+
+  // eslint-disable-next-line no-undefined
+  if (path !== null && path !== undefined && path !== '/') {
+    const parsedQuery = querystring.parse(path);
+    compilerName = parsedQuery.compilerName;
+  }
+})(__resourceQuery)
```
Comment (evilebottnawi) [inline client-src/default/index.js]: Move this to `util`

File: client-src/live/index.js [hunk 2]
```diff
@@ -60,6 +60,7 @@ $(() => {
       }
     },
     ok() {
+      console.log('OK!!');
```
Comment (evilebottnawi) [inline client-src/live/index.js]: Why you need `What is OK!!`? It is misleading need better text

File: lib/Server.js [hunk 3]
```diff
@@ -719,7 +719,7 @@ class Server {
         return;
       }
 
-      this._sendStats([connection], this.getStats(this._stats), true);
+      this._sendStats(null, [connection], this.getStats(this._stats), true);
```
Comment (evilebottnawi) [inline lib/Server.js]: null?

File: lib/utils/updateCompiler.js [hunk 4]
```diff
@@ -7,6 +7,13 @@
 const webpack = require('webpack');
 const addEntries = require('./addEntries');
 
+function guid() {
+  function s4() {
+    return ((1 + Math.random()) * 0x10000 | 0).toString(16).substring(1);
+  }
+  return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4();
+}
```
Comment (evilebottnawi) [inline lib/utils/updateCompiler.js]: Better use package for this

**Comment by orange4glace** [comment]
Agreed. Rather than modifying both server and client code, separating websocket connection like using channel by modifying only server code (and minimum client code modification as possible) seems to be a better implementation. I was kind of in a hurry so the implementation is actually dirty. I'll improve it with further commits :)

**Comment by orange4glace** [comment]
The new commit remains client code unchanged (so the code is completely same with the upstream master code) and only server code is changed. Please see the #2017 (even it contains tiny errors and corrected by the next commits)

I am trying to write and run tests, but running test keeps failing. It seems like Windows related error. I'm using VSCode in Windows 10 and if any code change happens with Windows environment editor, ESLint test goes fail, with unknown error.

Could you check if error happens on test with my repository source?

**Comment by evilebottnawi** [comment]
What is error(s) you got?

**Comment by orange4glace** [comment]
Actually the error is much same with what [Azure pipeline did](https://github.com/webpack/webpack-dev-server/pull/2017/checks?check_run_id=147933407). It seems error happens on **Lint:prettier**

Error log is [here](https://pastebin.com/kYrRkUjZ), and the complete log file is [here](https://pastebin.com/55y7WteG)

**Comment by evilebottnawi** [comment]
You can run tests using `npm run test:only` and avoid fixing linting issue right now, we can do this at last step of PR

**Comment by alexander-akait** [comment]
https://github.com/webpack/webpack-dev-server/issues/1503

**Pull request closed without merging**